app/connectors/connectors/sql-connector/src/main/java/io/syndesis/connector/sql/SqlConnectorMetaDataExtension.java

```diff
@@ -22,39 +22,54 @@
 import java.util.Map;
 import java.util.Optional;
 
+import io.syndesis.connector.sql.common.DatabaseMetaDataHelper;
+import io.syndesis.connector.sql.common.SqlStatementMetaData;
+import io.syndesis.connector.sql.common.SqlStatementParser;
+import org.apache.camel.CamelContext;
 import org.apache.camel.component.extension.metadata.AbstractMetaDataExtension;
 import org.apache.camel.component.extension.metadata.DefaultMetaData;
 
 public class SqlConnectorMetaDataExtension extends AbstractMetaDataExtension {
 
+    public SqlConnectorMetaDataExtension(CamelContext camelContext) {
+        super(camelContext);
+    }
+
     @Override
     public Optional<MetaData> meta(final Map<String, Object> properties) {
-
         final String sqlStatement = (String) properties.get("query");
 
-        if (sqlStatement!=null) {
-            try (Connection connection = DriverManager.getConnection(String.valueOf(properties.get("url")),
-                    String.valueOf(properties.get("user")), String.valueOf(properties.get("password")));) {
-
+        if (sqlStatement != null) {
+            try (Connection connection = createConnection(properties)) {
                 final DatabaseMetaData meta = connection.getMetaData();
-                final String defaultSchema = DatabaseMetaDataHelper.getDefaultSchema(
-                        meta.getDatabaseProductName(), String.valueOf(properties.get("user")));
+                final String defaultSchema = DatabaseMetaDataHelper.getDefaultSchema(meta.getDatabaseProductName(), String.valueOf(properties.get("user")));
                 final String schemaPattern = (String) properties.getOrDefault("schema-pattern", defaultSchema);
                 final SqlStatementParser parser = new SqlStatementParser(connection, schemaPattern, sqlStatement);
                 final SqlStatementMetaData sqlStatementMetaData = parseStatement(parser);
                 final MetaData metaData = new DefaultMetaData(null, null, sqlStatementMetaData);
-                return Optional.of(metaData);
 
+                return Optional.of(metaData);
             } catch (final SQLException e) {
                 throw new IllegalStateException(e);
             }
         } else {
-            final MetaData metaData = new DefaultMetaData(null, null, null);
-            return Optional.of(metaData);
+            return Optional.of(new DefaultMetaData(null, null, null));
         }
     }
+
+    // *********************************
+    // Helpers
+    // *********************************
     
-    protected  SqlStatementMetaData parseStatement(SqlStatementParser parser) throws SQLException {
+    private SqlStatementMetaData parseStatement(SqlStatementParser parser) throws SQLException {
         return parser.parse();
     }
+
+    private Connection createConnection(final Map<String, Object> properties) throws SQLException {
```

Comment:
we have this method in two classes perhaps we could refactor it into a `DatabaseHelper`?